### PR TITLE
folder-navigation

### PR DESCRIPTION
**Pull request by AllStackDev1**
Closes: #1395

# Create folder navigation

## Background

This is intended to be a reusable component, focused on building the structure for a given repository folder. It can contain any number of directories or files, with matching icons. When a user clicks on a folder, it will navigate into that folder. 

## Acceptance

- [x] users can navigate through a repository’s structure and view files
- [x] clicking on a folder navigates into that folder and shows any folders and files

### CODE REVIEW

File: solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            >
+              <div class="text-blue-600">..</div>
+            </A>
+          </Show>
+          <For each={tree()}>
+            {(item) => (
+              <div class={styles.cell}>
+                <div class="flex items-center">
+                  <div class="mr-2.5">
+                    {item.type === 'tree' ? (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx:70]:
```suggestion
                    <Switch fallback={<DocumentIcon class={styles.iconFile} />}>
                         <Match when={item.type === 'tree'}>
                              <FolderIcon class={styles.iconDir} />
                         </Match>
                    </Switch>
```

**Review by hdJerry — CHANGES_REQUESTED** [review]

File: solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      getRepoTree({
+        owner: local.owner,
+        name: local.name,
+        expression: `${local.branch}:${local.path || ''}`,
+      })
+  );
+
+  createEffect(() => {
+    if (resTree.isSuccess && !resTree.isLoading && resTree?.data.tree) {
+      setTree(parseQueryData(resTree?.data.tree));
```
Comment (Megio) [inline solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx:43]: We don't need the question mark after resTree in both line 42 and 43

File: solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    }`}
+                    data-testid={`file explorer list ${item.name}`}
+                    class={styles.link}
+                  >
+                    {item.name}
+                  </A>
+                </div>
+              </div>
+            )}
+          </For>
```
[the remark this answers is not in the document]
Comment (hdJerry) [reply solidstart-tanstackquery-tailwind-modules/src/components/FileExplorer/FileExplorer.tsx:88]: This has now been resolved. Thank sir

**Review by hdJerry — APPROVED** [review]
Thanks Mr.

**Review by Megio — APPROVED** [review]
🚀

**Pull request merged**